src/components/UpcomingEvents/index.jsx

```diff
@@ -1,79 +1,115 @@
-import React, { useState, useEffect } from 'react';
+import React from 'react';
 import './styles.css';
-import { charityAPI } from '../../clients';
+import { useCharityAPI } from '../../clients';
 import Heading from '../Heading';
-import { UpcomingEventsCardContainer } from '../UpcomingEventsCard';
+import { FeaturedCause } from '../FeaturedCause';
 import { parseISO, format } from 'date-fns';
 import { MainLoader, CardLoader } from './MyLoader';
 import { useInView } from 'react-intersection-observer';
 import { useSpring, animated } from 'react-spring';
 
-const Events = props => {
-  // Function to get add dates needed
-  function getDate(myDate) {
-    const theDate = parseISO(myDate);
+const UpcomingEventsSectionContainer = () => {
+  const { data, loading, dataError: error } = useCharityAPI('/upcoming-events');
+  const {
+    data: cardData,
+    loading: cardLoading,
+    dataError: cardError
+  } = useCharityAPI('/featured-cause');
+  return (
+    <UpcomingEventsSection
+      data={data}
+      loading={loading}
+      error={error}
+      cardData={cardData}
+      cardLoading={cardLoading}
+      cardError={cardError}
+    />
+  );
+};
+
+// Function to get add dates needed
+function getDate(myDate) {
+  const theDate = parseISO(myDate);
+
+  return {
+    time: format(theDate, 'hh:mm a').toLowerCase(),
+    day: format(theDate, 'dd'),
+    month: format(theDate, 'MMM'),
+    year: format(theDate, 'yyyy')
+  };
+}
+
+const Event = ({ data, loading }) => {
```

Comment:
El error hathandle nafsaha? 😂

Reply:
Dah component msh by fetch data bnafso .... fa eh elly 7ay error ?! law feeh error fa7ykoon fee el data kolaha we deeh et4amlaha handling fi el parent

Reply:
Mmmm.. I saw loading which indicates for me that it could load or error-ed.

why you pass the loading then?